sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -433,6 +450,7 @@ trait HashJoin extends BaseJoinExec with CodegenSupport {
     } else {
       ""
     }
+    val checkCondition = s"$numMatched.add(1);\n$conditionDef"
```

Comment:
Could you add `$numMatched.add(1);` directly before line 446 and line 451?

Reply:
Done. Fixed in the latest commit.